dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx

```diff
@@ -3,102 +3,487 @@
+  const paginationNumberFormatter = useCallback((params) => {
+    return '[' + params.value.toLocaleString() + ']';
+  }, []);
 
-        if (aValue < bValue) return column.direction === 'asc' ? -1 : 1;
-        if (aValue > bValue) return column.direction === 'asc' ? 1 : -1;
+  const selectDensityHandler = useCallback((value) => {
+    setSelectedRowDensity({ ...value });
+    gridRef.current.api.forEachNode((rowNode) => {
+      if (rowNode.data) {
+        rowNode.setRowHeight(value.height);
       }
-
-      return 0;
     });
-  }, [sortingColumns]);
+    gridRef.current.api.onRowHeightChanged();
+  }, []);
 
-  // Pagination
-  data = useMemo(() => {
-    const rowStart = pagination.pageIndex * pagination.pageSize;
-    const rowEnd = Math.min(rowStart + pagination.pageSize, data.length);
-    return data.slice(rowStart, rowEnd);
-  }, [data, pagination]);
+  const columnVisiblityHandler = useCallback((visible: boolean, feild: any) => {
+    const isExisting = columnVisibility.findIndex((i) => i === feild);
+    if (visible) {
+      if (isExisting > -1) {
+        columnVisibility.splice(isExisting, 1);
+        gridRef?.current?.columnApi?.setColumnsVisible([feild], true);
+      }
+    } else {
+      columnVisibility.push(feild);
+      gridRef?.current?.columnApi?.setColumnsVisible([feild], false);
+    }
+    setColumnVisibility([...columnVisibility]);
+  }, []);
 
-  const columns = fields.map((field) => {
-    return {
-      id: field.name,
+  // pagination
+  const pageCount = Math.ceil(raw_data.length / pageSize);
+  const [activePage, setActivePage] = useState(0);
+  const goToPage = ({ selected }: { selected: number }) => {
+    setActivePage(selected);
+    gridRef.current.api.paginationGoToPage(selected);
+    if (isFullScreen) {
+      gridRefFullScreen.current.api.paginationGoToPage(selected);
+    }
+  };
+
+  const setIsFullScreenHandler = (val: boolean) => {
+    setIsFullScreen(val);
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', hideGridFullScreenHandler);
+    return () => {
+      document.removeEventListener('keydown', hideGridFullScreenHandler);
     };
-  });
+  }, []);
 
-  // Column visibility
-  const [visibleColumns, setVisibleColumns] = useState(columns.map(({ id }) => id));
+  const hideGridFullScreenHandler = (e: any) => {
+    if (e.key === 'Escape') {
+      setIsFullScreen(false);
+    }
+  };
+  return (
+    <>
+      <GridHeader
+        isFullScreen={isFullScreen}
+        setIsFullScreenHandler={setIsFullScreenHandler}
+        selectedRowDensity={selectedRowDensity}
+        selectDensityHandler={selectDensityHandler}
+        columnVisiblityHandler={columnVisiblityHandler}
+        columns={columns}
+        columnVisibility={columnVisibility}
+      />
+      <div style={{ height: '460px' }}>
+        <AgGridReact
+          ref={gridRef}
+          rowData={raw_data}
+          columnDefs={columns}
+          defaultColDef={defaultColDef}
+          animateRows={true}
```

Comment:
I think, you do not need the true argument to be mentioned in a component.
`<AgGridReact animateRows={true} pagination={true} />` is same as `<AgGridReact animateRows pagination />`